services/src/northbound/src/main/java/org/openkilda/northbound/service/impl/LinkServiceImpl.java

```diff
@@ -108,72 +111,53 @@
+        }
 
-            for (CompletableFuture<InfoData> request : pendingRequest) {
-                LinkPropsResponse response = (LinkPropsResponse) request.join();
-
-                if (response.isSuccess()) {
-                    successCount += 1;
-                } else {
-                    errors.add(response.getError());
-                }
-            }
-            return new BatchResults(linkPropsList.size() - successCount, successCount, errors);
-        });
+        return collectResponses(pendingRequest, LinkPropsResponse.class)
+                .thenApply(responses -> getDelLinkPropsResult(responses, errors));
     }
 
     @Override
     public CompletableFuture<BatchResults> delLinkProps(List<LinkPropsDto> linkPropsList) {
-        return CompletableFuture.supplyAsync(() -> {
-            List<CompletableFuture<List<InfoData>>> pendingRequest = new ArrayList<>(linkPropsList.size());
+        List<CompletableFuture<List<InfoData>>> pendingRequest = new ArrayList<>(linkPropsList.size());
 
-            for (LinkPropsDto requestItem : linkPropsList) {
-                LinkPropsDrop teRequest = new LinkPropsDrop(linkPropsMapper.toLinkPropsMask(requestItem));
-                String requestId = idFactory.produceChained(RequestCorrelationId.getId());
-                CommandMessage message = new CommandMessage(teRequest, System.currentTimeMillis(), requestId);
+        for (LinkPropsDto requestItem : linkPropsList) {
+            LinkPropsDrop teRequest = new LinkPropsDrop(linkPropsMapper.toLinkPropsMask(requestItem));
+            String requestId = idFactory.produceChained(RequestCorrelationId.getId());
+            CommandMessage message = new CommandMessage(teRequest, System.currentTimeMillis(), requestId);
 
-                pendingRequest.add(messagingChannel.sendAndGetChunked(topologyEngineTopic, message));
-            }
-            return pendingRequest;
-        }).thenApply(pendingRequests -> {
-            int successCount = 0;
-            ArrayList<String> errors = new ArrayList<>();
-            for (CompletableFuture<List<InfoData>> request : pendingRequests) {
-                List<InfoData> responseBatch = request.join();
-                for (InfoData response : responseBatch) {
-
-                    LinkPropsResponse linkProps = (LinkPropsResponse) response;
-                    if (linkProps.isSuccess()) {
-                        successCount += 1;
-                    } else {
-                        errors.add(linkProps.getError());
-                    }
-                }
+            pendingRequest.add(messagingChannel.sendAndGetChunked(topologyEngineTopic, message));
+        }
+
+        return collectChunkedResponses(pendingRequest, LinkPropsResponse.class)
+                .thenApply(responses -> getDelLinkPropsResult(responses, new ArrayList<>()));
+    }
+
+    private BatchResults getDelLinkPropsResult(List<LinkPropsResponse> responses, List<String> errors) {
```

Comment:
this method is uses in "set" and in "del" operations, it should not containt "del" chunk in it.